src/pykiso/auxiliary.py

```diff
@@ -212,7 +224,8 @@ def _start_rx_task(self) -> None:
         with self.rx_lock:
             task_name = f"{self.name}_rx"
             log.internal_debug("start reception task %s", task_name)
-            self.rx_thread = threading.Thread(name=task_name, target=self._reception_task)
+            # Any created thread should disappear after main-thread exit
+            self.rx_thread = threading.Thread(name=task_name, target=self._reception_task, daemon=True)
```

Comment:
We should definitely test the different auxiliaries thoughtfully before merging this change, switching to daemon threads could have quite a lot of unexpected consequences (and if the auxiliries are "killed" at the end, we'll probably lose some insights on the correctness of our thread implementations)

Reply:
@sebclrsn 
When I start the context manager with nothing in, it sometimes hang. From what I have seen, the proxy does not exit correctly. I'm not sure on how to resolve this issue. Maybe you can take a look at it?